src/tribler-core/tribler_core/modules/bandwidth_accounting/history.py

```diff
@@ -0,0 +1,17 @@
+from pony.orm import PrimaryKey, Required
+
+
+def define_binding(bandwidth_database):
+    db = bandwidth_database.database
+
+    class BandwidthHistory(db.Entity):
+        """
+        This ORM class represents a mutation of ones bandwidth balance.
+        We store the last 100 mutations in ones bandwidth balance.
+        """
+
+        rowid = PrimaryKey(int, auto=True)
+        timestamp = Required(int, size=64)
```

Comment:
Looking at queries, I think the following indexes would be useful:
* composite_key(public_key_a, public_key_b), for searching by pair of (key_a, key_b) as well as just key_a;
* public_key_b
* timestamp (for ordering)

I don't see balance mentions in query conditions, so probably it is better not to create an index for it, as each unnecessary index slowdowns inserts a bit.

Reply:
Since we only store the last 100 balance mutations (by default), the number of rows should be small. @kozlovsky do you think it would still be worth to add indices?

Reply:
For such a small number of rows, indices are unnecessary, as the number of database pages we read will be very small anyway.

Reply:
Ok thanks! 👍